### PR TITLE
Update header and content based routing templates

### PR DESCRIPTION
**Pull request by indikasampath2000**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: plugins/org.wso2.developerstudio.eclipse.esb.dashboard.templates/Samples/ContentBasedRoutingTemplate/src/main/synapse-config/proxy-services/ArithmaticOperationService.xml [hunk 1]
```diff
@@ -3,33 +3,55 @@
     <target>
         <inSequence>
             <log description="Log Request Payload" level="full"/>
-            <switch description="Retrieve the first element name and match in the case. Otherwise, send back custom payload." source="local-name(//soapenv:Body/*[1])" xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
-                <case regex="AddInteger">
-                    <header name="Action" scope="default" value="http://tempuri.org/SOAP.Demo.AddInteger"/>
-                    <send description="Send request to the endpoint">
+            <switch description="Route based on the Operation" source="//ArithmaticOperation/Operation" xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
```
Comment (isudana) [inline plugins/org.wso2.developerstudio.eclipse.esb.dashboard.templates/Samples/ContentBasedRoutingTemplate/src/main/synapse-config/proxy-services/ArithmaticOperationService.xml]: do we need this namespace ?

File: plugins/org.wso2.developerstudio.eclipse.esb.dashboard.templates/Samples/ContentBasedRoutingTemplate/src/main/synapse-config/proxy-services/ArithmaticOperationService.xml [hunk 2]
```diff
@@ -3,33 +3,55 @@
...
+                <case regex="Add">
+                    <payloadFactory description="Build backend payload" media-type="xml">
```
Comment (isudana) [inline plugins/org.wso2.developerstudio.eclipse.esb.dashboard.templates/Samples/ContentBasedRoutingTemplate/src/main/synapse-config/proxy-services/ArithmaticOperationService.xml]: Shall we say "construct payload for addition operation"

File: plugins/org.wso2.developerstudio.eclipse.esb.dashboard.templates/Samples/ContentBasedRoutingTemplate/src/main/synapse-config/proxy-services/ArithmaticOperationService.xml [hunk 3]
```diff
@@ -3,33 +3,55 @@
...
+                    <call description="Send request to the endpoint">
                         <endpoint key="NumberAdditionEP"/>
-                    </send>
+                    </call>
                 </case>
-                <case regex="DivideInteger">
-                    <header name="Action" scope="default" value="http://tempuri.org/SOAP.Demo.DivideInteger"/>
-                    <send>
+                <case regex="Divide">
+                    <payloadFactory description="Build backend payload" media-type="xml">
```
Comment (isudana) [inline plugins/org.wso2.developerstudio.eclipse.esb.dashboard.templates/Samples/ContentBasedRoutingTemplate/src/main/synapse-config/proxy-services/ArithmaticOperationService.xml]: Shall we say "construct payload for division operation"

File: plugins/org.wso2.developerstudio.eclipse.esb.dashboard.templates/Samples/ContentBasedRoutingTemplate/src/main/synapse-config/proxy-services/ArithmaticOperationService.xml [hunk 4]
```diff
@@ -3,33 +3,55 @@
...
+                        </args>
+                    </payloadFactory>
+                    <header description="Set SOAPAction header" name="Action" scope="default" value="http://tempuri.org/SOAP.Demo.DivideInteger"/>
+                    <call>
                         <endpoint key="NumberDivisionEP"/>
-                    </send>
+                    </call>
                 </case>
                 <default>
                     <payloadFactory description="Build custom payload to send back" media-type="xml">
```
Comment (isudana) [inline plugins/org.wso2.developerstudio.eclipse.esb.dashboard.templates/Samples/ContentBasedRoutingTemplate/src/main/synapse-config/proxy-services/ArithmaticOperationService.xml]: Shall we say "create response payload for unsupported operation"

File: plugins/org.wso2.developerstudio.eclipse.esb.dashboard.templates/Samples/HeaderBasedRoutingTemplate/src/main/synapse-config/api/CalculatorAPI.xml [hunk 5]
```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (isudana) [inline plugins/org.wso2.developerstudio.eclipse.esb.dashboard.templates/Samples/HeaderBasedRoutingTemplate/src/main/synapse-config/api/CalculatorAPI.xml]: same comments for ArithmaticOperationService.xml applies here

File: plugins/org.wso2.developerstudio.eclipse.esb.dashboard.templates/Samples/ContentBasedRoutingTemplate/src/main/synapse-config/proxy-services/ArithmaticOperationService.xml (hunk 1 above)
Comment (indikasampath2000) [reply plugins/org.wso2.developerstudio.eclipse.esb.dashboard.templates/Samples/ContentBasedRoutingTemplate/src/main/synapse-config/proxy-services/ArithmaticOperationService.xml]: Fixed.

File: plugins/org.wso2.developerstudio.eclipse.esb.dashboard.templates/Samples/ContentBasedRoutingTemplate/src/main/synapse-config/proxy-services/ArithmaticOperationService.xml (hunk 2 above)
Comment (indikasampath2000) [reply plugins/org.wso2.developerstudio.eclipse.esb.dashboard.templates/Samples/ContentBasedRoutingTemplate/src/main/synapse-config/proxy-services/ArithmaticOperationService.xml]: Fixed.

File: plugins/org.wso2.developerstudio.eclipse.esb.dashboard.templates/Samples/ContentBasedRoutingTemplate/src/main/synapse-config/proxy-services/ArithmaticOperationService.xml (hunk 3 above)
Comment (indikasampath2000) [reply plugins/org.wso2.developerstudio.eclipse.esb.dashboard.templates/Samples/ContentBasedRoutingTemplate/src/main/synapse-config/proxy-services/ArithmaticOperationService.xml]: Fixed.

File: plugins/org.wso2.developerstudio.eclipse.esb.dashboard.templates/Samples/ContentBasedRoutingTemplate/src/main/synapse-config/proxy-services/ArithmaticOperationService.xml (hunk 4 above)
Comment (indikasampath2000) [reply plugins/org.wso2.developerstudio.eclipse.esb.dashboard.templates/Samples/ContentBasedRoutingTemplate/src/main/synapse-config/proxy-services/ArithmaticOperationService.xml]: Fixed.

File: plugins/org.wso2.developerstudio.eclipse.esb.dashboard.templates/Samples/HeaderBasedRoutingTemplate/src/main/synapse-config/api/CalculatorAPI.xml (hunk 5 above)
Comment (indikasampath2000) [reply plugins/org.wso2.developerstudio.eclipse.esb.dashboard.templates/Samples/HeaderBasedRoutingTemplate/src/main/synapse-config/api/CalculatorAPI.xml]: Fixed.

**Pull request merged**